### DISCUSSION TITLE
Prevent changing the base branch?

### DISCUSSION THREAD

**Discussion by hugovk** [community/community#56333] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

For PRs, is it possible to prevent changing the base branch?

![image](https://github.com/community/maintainers/assets/1324225/1d9ae6bf-6f99-423e-ae99-ba1001f7984b)

On the https://github.com/python/cpython project, we sometimes get PRs opened against older branches (e.g. `3.11`) when our workflow is to open against `main` and then backport to older branches (we have bots).

Sometimes a new contributor accidentally opens a PR against an old branch like `3.11`, and when we ask for a fresh PR, they instead change the base branch to `main` (and why not, this usually works for small repos).

But because our branches have diverged so much, it requests reviews from _lots_ of `CODEOWNERS` which essentially spams a load of maintainers.

For example: https://github.com/python/cpython/pull/104498

<details>
<summary>screenshot</summary>

![image](https://github.com/community/maintainers/assets/1324225/076c42f3-5fd9-46cb-937a-aeb3d62201a3)

</details>

If we could disable changing the base branch for the repo, this would help prevent this. Thanks!

**Comment by Jackenmen** [comment]
I've just done the same thing, *also* on CPython in https://github.com/python/cpython/pull/105043. Personally, I think that a better solution would be for this dialog to more accurately describe what is going to happen after changing the branch:
![image](https://github.com/community/community/assets/6032823/f90416dd-a67b-4623-b5d5-4196e1183eb2)

I think it would help if GitHub could calculate what's going to happen after changing the branch (number of commits that you'll be proposing after changing the branch, number of people you'll request a review from, diff size, and anything else that could be useful) and present it to the user with an appropriate "are you sure you *really* want to do this?" confirmation.
